packages/store/ts/config/experimental/resolveConfig.test-d.ts

```diff
@@ -0,0 +1,51 @@
+import { describe, expectTypeOf } from "vitest";
+import { mudConfig } from "../../register/mudConfig";
+import { resolveConfig } from "./resolveConfig";
+
+const config = resolveConfig(
+  mudConfig({
+    // Seems like we need `as const` here to keep the strong type.
+    // Note it resolves to the strong `""` type if no namespace is provided.
+    // TODO: require the entire input config to be `const`
```

Comment:
I remember trying to add `as const` to our default MUD configs (world config, store config) and it broke downstream types in weird ways. Hopefully we can just add this more specific `as const` to just the namespace without those downstream effects, but yeah, should aim to `as const` the whole config.

Reply:
yeah the arrays for e.g. `enums` currently complain if the whole config is `as const`

Reply:
Do we wanna add `as const` to the world/store namespaces?